cg/meta/invoice.py

```diff
@@ -14,15 +14,35 @@ def __init__(self, db: Store, lims_api: lims.LimsAPI, invoice_obj: models.Invoic
         self.log = []
         self.invoice_obj = invoice_obj
         self.customer_obj = invoice_obj.customer
+        self.record_type = ''
+        self.raw_records = []
+        self._get_record_type()
+
+
+    def _get_record_type(self):
+        """Define the record_type based on the invoice object. 
+        It can only be either pool, sample or microbial"""
+
+        if self.invoice_obj.pools:
+            self.raw_records = self.invoice_obj.pools
+            self.record_type = 'Pool'
+        elif self.invoice_obj.samples:
+            self.record_type = 'Sample'
+            self.raw_records = self.invoice_obj.samples
+        elif self.invoice_obj.microbial_samples:
+            self.record_type = 'Microbial'
+            self.raw_records = self.invoice_obj.microbial_samples
+
 
     def prepare_contact_info(self, costcenter):
+        """Function to prepare contact info for a customer"""
+
         msg = f'Could not open/generate invoice. Contact information missing in database for ' \
-            f'customer {self.customer_obj.internal_id}. See log files.'
-        if costcenter.lower() == 'kth':
-            customer = self.db.customer('cust999')
-        else:
-            customer = self.customer_obj
+              f'customer {self.customer_obj.internal_id}. See log files.'
+
+        customer = self.db.customer('cust999') if costcenter.lower() == 'kth' else self.customer_obj
```

Comment:
Something to think about for a later iteration: never hardcode. A possible solution is to add an extra field to the customer table denoting that this customer is special in the way 'cust999' is special. That way you can query it, exclude it from queries, etc...

No need to refactor that now.